lib/Sema/CSSimplify.cpp

```diff
@@ -1220,6 +1226,18 @@ ConstraintSystem::TypeMatchResult constraints::matchCallArguments(
     if (!callArgumentMatch)
       return cs.getTypeMatchFailure(locator);
 
+    // When checking to see if the callable value can be explicit called to
+    // produce the expected type, we need to make sure that we fail here for
+    // missing argument (which normally have fix recorded in argument tracker)
+    // except if all parameters have a default values.
```

Comment:
> I think the idea where would be to suggest adding a call and missing arguments, no?

Not really sure, that would mean a fix for a declaration like `func callAsFunction(a: Int) -> Int { 0 }` would be `(a: <#editor placeholder#>)`? 

>  I guess we'd just need to adjust a fix for synthesized arguments to handle implicit calls to support that...

I didn't get it, have to take a look into this synthesized fix first :)

Reply:
bq. Not really sure, that would mean a fix for a declaration like `func callAsFunction(a: Int) -> Int { 0 }` would be `(a: <#editor placeholder#>)`?

Yeah, that's exactly how I see it.

Reply:
Ahh, I thought about this at the beginning, but as I was looking at the existing `InsertExplicitCall` fix for function types and since it didn't have this, I figure it would be something more simple to maintain this behavior here. Do you think is worth introducing this here for both? 
Also, in the beginning, I was trying to think from the user perspective that to insert a `()` call in value mismatch is more likely to be an expected fix than a call with parameters, so the more generic mismatch fix would fit better in this 'with params' case? I'm not sure, just trying to think if insert explicit call would be a better diagnostic in this case :)

Reply:
This is going to be tricky because we'd effectively have to combine (or coalesce) missing arguments with missing call fix to make it work. I think we'd need some test cases to make sure that if this happens diagnostic logic wouldn't crash since there is no call :)

Reply:
Yeah, I was also thinking about some other cases where we should give a good thought about like 
```
struct S {
   func callAsFunction(a: Int) -> Int { 0 } 
   func callAsFunction(b: String) -> Int { 0 } 
}
```
What we should do in a mismatch`S` and `Int` like suggest one or another doesn't feel right... 
So offer two fix-its? In that case, diagnostic wording should probably change in those situations.

Reply:
I think in this case, if there are no other options, diagnostic should just say that call is missing and let the developer disambiguate...